Regression in 0.11.22: `uv venv` writes truncated `version_info` to `pyvenv.cfg`, breaking pre-commit health check

### Summary
`uv 0.11.22` writes a truncated `version_info` into `pyvenv.cfg` (missing the `.final.0` release-level suffix that `venv` / `virtualenv` produce). When pre-commit creates its hook environments via `pre-commit-uv` (i.e. `uv venv`), pre-commit's post-install health check compares the recorded version against the interpreter's full version string, finds a mismatch, and rejects the environment — aborting the run before any hook executes. This is a regression: `uv 0.11.21` worked. Looks like a re-introduction of #1689 ("We now match `venv` when reporting the Python version").

### Regression (bisected via CI)
Identical project, runner image, reusable workflow (same commit), `pre-commit-uv 4.2.1`, and managed `cpython-3.13.14`. The only variable that changed is the uv version (CI resolves uv to latest each run):

- 2026-06-18, uv **0.11.21** -> pre-commit passes (all hooks green)
- 2026-06-19, uv **0.11.22** -> pre-commit fails at the health check

### Observed failure (uv 0.11.22)
Step: `uv tool run --with pre-commit-uv pre-commit run --show-diff-on-failure --color=always --all-files`, `UV_MANAGED_PYTHON=1`, ubuntu-22.04.

```
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
[INFO] Using pre-commit with uv 0.11.22 via pre-commit-uv 4.2.1
An unexpected error has occurred: AssertionError: BUG: expected environment for python to be healthy immediately after install, please open an issue describing your environment

more info:

virtualenv python version did not match created version:
- actual version: 3.13.14
- expected version: 3.13

Check the log at /home/runner/.cache/pre-commit/pre-commit.log
Error: Process completed with exit code 3.
```

`actual version` is the real interpreter (`3.13.14`); `expected version` is what was recorded in `pyvenv.cfg`.

### Minimal reproduction
`uv venv` writes a truncated `version_info` compared to `virtualenv`:

```
$ uvx virtualenv /tmp/vetest && grep version_info /tmp/vetest/pyvenv.cfg
version_info = 3.13.0.final.0

$ uv venv /tmp/uvtest --python 3.13.0 && grep version_info /tmp/uvtest/pyvenv.cfg
Using CPython 3.13.0 interpreter at: ~/.pyenv/versions/3.13.0/bin/python3.13
version_info = 3.13.0
```

### Expected vs actual
- Expected: `version_info` in `X.Y.Z.final.0` form (matching `venv` / `virtualenv`).
- Actual: truncated (`3.13.0` in the local repro; reported as `3.13` by pre-commit in CI). Either way it diverges from the format pre-commit expects, so the health check fails.

### Environment
- uv: 0.11.22 (regressed); 0.11.21 worked
- pre-commit-uv: 4.2.1
- OS: ubuntu-22.04 (CI), macOS arm64 (local repro)
- Related: #1689

--- a/crates/uv-virtualenv/src/virtualenv.rs
+++ b/crates/uv-virtualenv/src/virtualenv.rs
@@ -528,11 +528,7 @@ pub(crate) fn create(
         ("uv".to_string(), version().to_string()),
         (
             "version_info".to_string(),
-            if using_minor_version_link {
-                interpreter.python_minor_version().to_string()
-            } else {
-                interpreter.markers().python_full_version().string.clone()
-            },
+            interpreter.markers().python_full_version().string.clone(),
         ),
         (
             "include-system-site-packages".to_string(),

--- a/crates/uv/tests/python/python_upgrade.rs
+++ b/crates/uv/tests/python/python_upgrade.rs
@@ -2,7 +2,6 @@ use anyhow::Result;
 use assert_cmd::assert::OutputAssertExt;
 use assert_fs::fixture::{FileTouch, FileWriteStr};
 use assert_fs::prelude::PathChild;
-use insta::assert_snapshot;
 use uv_python::managed::platform_key_from_env;
 use uv_static::EnvVars;
 use uv_test::{LATEST_PYTHON_3_12, uv_snapshot};
@@ -840,73 +839,6 @@ fn python_upgrade_build_version() {
     ");
 }
 
-// A project environment should survive a transparent Python patch upgrade.
-#[test]
-fn python_sync_transparent_patch_upgrade_reuses_environment() -> Result<()> {
-    let context = uv_test::test_context_with_versions!(&[])
-        .with_python_download_cache()
-        .with_filtered_python_keys()
-        .with_filtered_exe_suffix()
-        .with_managed_python_dirs()
-        .with_filtered_latest_python_versions()
-        .with_pyvenv_cfg_filters();
-
-    context.temp_dir.child("pyproject.toml").write_str(
-        r#"
-        [project]
-        name = "project"
-        version = "0.1.0"
-        requires-python = ">=3.10"
-        "#,
-    )?;
-
-    context.python_install().arg("3.10.17").assert().success();
-    context
-        .sync()
-        .arg("--python")
-        .arg("3.10")
-        .assert()
-        .success();
-
-    // The minor-only `version_info` allows compatibility checks to accept a newer patch.
-    let pyvenv_cfg = fs_err::read_to_string(context.venv.child("pyvenv.cfg"))?;
-    insta::with_settings!({ filters => context.filters() }, {
-        assert_snapshot!(pyvenv_cfg, @r"
-        home = [PYTHON_HOME]
-        implementation = CPython
-        uv = [UV_VERSION]
-        version_info = 3.10
-        include-system-site-packages = false
-        prompt = project
-        ");
-    });
-
-    context.python_upgrade().arg("3.10").assert().success();
-
-    // The sync should not report removing or creating the environment.
-    uv_snapshot!(context.filters(), context.sync().arg("--python").arg("3.10"), @r"
-    success: true
-    exit_code: 0
-    ----- stdout -----
-
-    ----- stderr -----
-    Resolved 1 package in [TIME]
-    Checked in [TIME]
-    ");
-
-    // The reused environment should run the upgraded Python patch.
-    uv_snapshot!(context.filters(), context.python_command().arg("--version"), @r"
-    success: true
-    exit_code: 0
-    ----- stdout -----
-    Python 3.10.[LATEST]
-
-    ----- stderr -----
-    ");
-
-    Ok(())
-}
-
 // Regression test for <https://github.com/astral-sh/uv/issues/19100>.
 //
 // A virtual environment created by `uv sync` should pin to a patch version
@@ -920,8 +852,7 @@ fn python_sync_honors_pinned_patch_version() -> Result<()> {
         .with_filtered_python_keys()
         .with_filtered_exe_suffix()
         .with_managed_python_dirs()
-        .with_filtered_latest_python_versions()
-        .with_pyvenv_cfg_filters();
+        .with_filtered_latest_python_versions();
 
     let pyproject_toml = context.temp_dir.child("pyproject.toml");
     pyproject_toml.write_str(
@@ -966,19 +897,6 @@ fn python_sync_honors_pinned_patch_version() -> Result<()> {
     Resolved 1 package in [TIME]
     Checked in [TIME]
     ");
-
-    // An exact-patch environment must retain the patch in `version_info`.
-    let pyvenv_cfg = fs_err::read_to_string(context.venv.child("pyvenv.cfg"))?;
-    insta::with_settings!({ filters => context.filters() }, {
-        assert_snapshot!(pyvenv_cfg, @r"
-        home = [PYTHON_HOME]
-        implementation = CPython
-        uv = [UV_VERSION]
-        version_info = 3.10.17
-        include-system-site-packages = false
-        prompt = project
-        ");
-    });
 
     // Install a newer patch version, which moves the mutable `cpython-3.10`
     // minor-version link to the newer release.

